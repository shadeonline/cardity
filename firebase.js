// Import the functions you need from the SDKs you need
import { initializeApp } from "firebase/app";
import { getAuth, createUserWithEmailAndPassword, signInWithEmailAndPassword} from 'firebase/auth';
// TODO: Add SDKs for Firebase products that you want to use
// https://firebase.google.com/docs/web/setup#available-libraries

// Your web app's Firebase configuration
// For Firebase JS SDK v7.20.0 and later, measurementId is optional
const firebaseConfig = {
  apiKey: "AIzaSyCf2BsSO7intMEZyZQUbf3gdQK_TNOvxHE",
  authDomain: "carditydb.firebaseapp.com",
  projectId: "carditydb",
  storageBucket: "carditydb.appspot.com",
  messagingSenderId: "627053711755",
  appId: "1:627053711755:web:caa91da4f88ef749419f8e",
  measurementId: "G-44WRTNPMRV"
};

// Initialize Firebase
const app = initializeApp(firebaseConfig);
const auth = getAuth(app);

export { auth};