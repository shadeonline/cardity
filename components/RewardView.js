import React, { useEffect, useState } from 'react';
import { Text, View, Image, TouchableOpacity } from 'react-native';
import { collection, getDocs, doc, getDoc, updateDoc, arrayUnion } from 'firebase/firestore';
import { auth, firestore, firebase } from '../firebase';
import { useIsFocused } from '@react-navigation/native'

const RewardView = ({ card }) => {

  const isFocused = useIsFocused()
  useEffect(() => {
    if (isFocused) {
      console.log("Reward");
    }
  }, [isFocused]);


  const rewards = [
    {
      id: 1,
      title: 'Reward 1',
      description: 'Description of reward 1',
    },
    {
      id: 2,
      title: 'Reward 2',
      description: 'Description of reward 2',
    },
    {
      id: 3,
      title: 'Reward 3',
      description: 'Description of reward 3',
    },
  ];

  const handleClaimReward = (reward) => {
    // Logic to handle the reward claim
    console.log('Claimed reward:', reward);
  };

  return (
    <View style={styles.tabContainer}>
      <Text style={styles.heading}>Claimable Rewards</Text>
      {rewards.map((reward) => (
        <TouchableOpacity
          key={reward.id}
          style={styles.rewardContainer}
          onPress={() => handleClaimReward(reward)}
        >
          <View style={styles.rewardContent}>
            <Text style={styles.rewardTitle}>{reward.title}</Text>
            <Text style={styles.rewardDescription}>{reward.description}</Text>
          </View>
        </TouchableOpacity>
      ))}
    </View>
  );
};

const styles = {
  tabContainer: {
    flex: 1,
    backgroundColor: 'white',
    paddingTop: 20,
    paddingHorizontal: 20,
  },
  heading: {
    fontSize: 24,
    marginBottom: 10,
    textAlign: 'center',
  },
  rewardContainer: {
    backgroundColor: '#e6e6e6',
    borderRadius: 8,
    marginBottom: 10,
  },
  rewardContent: {
    padding: 10,
  },
  rewardTitle: {
    fontSize: 18,
    fontWeight: 'bold',
  },
  rewardDescription: {
    fontSize: 16,
    marginTop: 5,
  },
};

export default RewardView;
