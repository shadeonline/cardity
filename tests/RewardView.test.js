import React from 'react';
import { render } from 'react-native-testing-library';
import RewardView from '../components/RewardView';
import { auth } from '../firebase';
import { NavigationContainer } from '@react-navigation/native';

jest.mock('../firebaseFunctions', () => ({
  firebaseFetchUserProfileReward: jest.fn(),
}));

jest.mock('../firebase', () => ({
  auth: {
    currentUser: {
      uid: 'mockedUserId',
    },
  },
}));

jest.mock('@react-navigation/stack', () => {
  const actualStack = jest.requireActual('@react-navigation/stack');
  return {
    ...actualStack,
    createStackNavigator: () => ({
      Navigator: ({ children }) => <>{children}</>,
      Screen: ({ children }) => <>{children}</>,
    }),
  };
});

describe('RewardView Component', () => {
  it('matches snapshot', () => {
    // Mock card data
    const card = {
      "cardId": 992847266036,
      "cardName": "The Ice Cream Card",
      "color": undefined,
      "image": "icecreamRestaurant",
      "loyaltyCard": "loyaltyCards/6xHLagIWsk8Z8Z4ndbO4",
      "loyaltyProgram": "loyaltyPrograms/kAdCUvzu63ybzK09VXEI",
      "progress": 5
    };

    // Render the RewardView component with the mock data
    const { toJSON } = render(
      <NavigationContainer>
        <RewardView card={card} />
      </NavigationContainer>);

    // Use Jest's expect to match the snapshot
    expect(toJSON()).toMatchSnapshot();
  });
});
