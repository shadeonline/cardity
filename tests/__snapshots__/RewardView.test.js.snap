// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RewardView Component matches snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "flex": 1,
      "paddingHorizontal": 20,
      "paddingTop": 20,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 28,
        "fontWeight": "bold",
        "marginBottom": 20,
        "textAlign": "center",
      }
    }
  >
    Claimable Rewards
  </Text>
  <RCTScrollView>
    <View />
  </RCTScrollView>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  />
</View>
`;
